x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/GetSearchApplicationAction.java

```diff
@@ -99,14 +128,46 @@ public Response(
             long updatedAtMillis,
             SearchApplicationTemplate template
         ) {
+            if (Strings.isNullOrEmpty(name)) {
```

Comment:
Shouldn't this check be done in the `SearchApplication` constructor?

Reply:
I think I see why @saarikabhasi put this check here, since it's the individual constructor response. I don't think it's needed though since we already do that validation in `SearchApplication`?